From 77e9cfe912adf36b74f80cbae937c42d2d5973fe Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@suse.de>
Date: Fri, 27 Feb 2009 20:28:04 -0800
Subject: Staging: line6: remove PT_REGS

As the code is in the kernel tree, it's no longer needed.

Cc: Markus Grabner <grabner@icg.tugraz.at>
Cc: Mariusz Kozlowski <m.kozlowski@tuxland.pl>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/line6/capture.c  |    2 +-
 drivers/staging/line6/config.h   |   10 ----------
 drivers/staging/line6/driver.c   |    6 +++---
 drivers/staging/line6/midi.c     |    2 +-
 drivers/staging/line6/playback.c |    2 +-
 5 files changed, 6 insertions(+), 16 deletions(-)

--- a/drivers/staging/line6/capture.c
+++ b/drivers/staging/line6/capture.c
@@ -136,7 +136,7 @@ void unlink_wait_clear_audio_in_urbs(str
 /*
 	Callback for completed capture URB.
 */
-static void audio_in_callback(struct urb *urb PT_REGS)
+static void audio_in_callback(struct urb *urb)
 {
 	int i, index, length = 0, shutdown = 0;
 	int frames;
--- a/drivers/staging/line6/config.h
+++ b/drivers/staging/line6/config.h
@@ -43,16 +43,6 @@
 #define CHECKPOINT printk("line6usb: %s (%s:%d)\n", __FUNCTION__, __FILE__, __LINE__)
 #endif
 
-/**
-   In Linux 2.6.20 and later, the pt_regs is no longer passed to USB callback
-   functions.
-*/
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 20)
-#define PT_REGS
-#else
-#define PT_REGS , struct pt_regs *regs
-#endif
-
 #if DO_DEBUG_MESSAGES
 #define DEBUG_MESSAGES(x) (x)
 #else
--- a/drivers/staging/line6/driver.c
+++ b/drivers/staging/line6/driver.c
@@ -94,7 +94,7 @@ struct message
 /*
 	Forward declarations.
 */
-static void line6_data_received(struct urb *urb PT_REGS);
+static void line6_data_received(struct urb *urb);
 static int line6_send_raw_message_async_part(struct message *msg, struct urb *urb);
 
 
@@ -202,7 +202,7 @@ int line6_send_raw_message(struct usb_li
 /*
 	Notification of completion of asynchronous request transmission.
 */
-static void line6_async_request_sent(struct urb *urb PT_REGS)
+static void line6_async_request_sent(struct urb *urb)
 {
 	struct message *msg = (struct message *)urb->context;
 
@@ -315,7 +315,7 @@ char *line6_alloc_sysex_buffer(struct us
 /*
 	Notification of data received from the Line6 device.
 */
-static void line6_data_received(struct urb *urb PT_REGS)
+static void line6_data_received(struct urb *urb)
 {
 	struct usb_line6 *line6 = (struct usb_line6 *)urb->context;
 	struct MidiBuffer *mb = &line6->line6midi->midibuf_in;
--- a/drivers/staging/line6/midi.c
+++ b/drivers/staging/line6/midi.c
@@ -87,7 +87,7 @@ static void line6_midi_transmit(struct s
 /*
 	Notification of completion of MIDI transmission.
 */
-static void midi_sent(struct urb *urb PT_REGS)
+static void midi_sent(struct urb *urb)
 {
 	unsigned long flags;
 	int status;
--- a/drivers/staging/line6/playback.c
+++ b/drivers/staging/line6/playback.c
@@ -221,7 +221,7 @@ void unlink_wait_clear_audio_out_urbs(st
 /*
 	Callback for completed playback URB.
 */
-static void audio_out_callback(struct urb *urb PT_REGS)
+static void audio_out_callback(struct urb *urb)
 {
 	int i, index, length = 0, shutdown = 0;
 	unsigned long flags;
